src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,1017 @@
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  vector<vector<double>> output;
+  TfIdfEncoding<CharExtract::TokenType> encoder;
+
+  // Passing a empty string to encode characters
+  encoder.Encode(input, output, CharExtract());
+  vector<vector<double>> expected = {
+    { 1.2876820724517808, 2, 2.5753641449035616, 0, 0 },
+    { 0, 2, 2.5753641449035616, 2.5753641449035616, 1.6931471805599454 },
+    { 1.2876820724517808, 1, 0, 1.2876820724517808, 0 }
+  };
+  for (size_t i = 0; i < expected.size(); i++)
+    for (size_t j = 0; j < expected[i].size(); j++)
+      BOOST_REQUIRE_CLOSE(expected[i][j], output[i][j], 1e-12);
+}
+
+/**
+ * Test the Tf-Idf Encoding using rawcount type and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(TfIdfRawCountEncodingTest)
+{
+  using DictionaryType = StringEncodingDictionary<boost::string_view>;
+
+  arma::mat output;
+  TfIdfEncoding<SplitByAnyOf::TokenType> encoder(
+      (TfIdfEncodingPolicy(0, false)));
+  SplitByAnyOf tokenizer(" ");
+
+  encoder.Encode(stringEncodingInput, output, tokenizer);
+
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+
+  arma::mat expected = {
+    {  1.40546510810816, 1.40546510810816, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 1, 2.09861228866811, 1.40546510810816,
+       1.40546510810816, 1.40546510810816, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 1, 2.09861228866811, 2.09861228866811, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0  },
+    {  0, 1.40546510810816, 0, 0, 0, 2, 0, 0, 4.21639532432449,
+       4.21639532432449, 0, 0, 0, 3, 0, 0, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 6.29583686600433, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  },
+    {  1.40546510810816, 0, 0, 0, 0, 1, 0, 1.40546510810816, 0, 0, 0, 0, 0, 1,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811  }
+  };
+  CheckMatrices(output, expected, 1e-12);
+}
+
+/**
+ * Test the one pass modification of the TfIdf encoding algorithm, with rawcount
+ * as type, but with smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(OnePassTfIdfRawCountEncodingTest)
```

Comment:
It's not the one pass modification. It's exactly the same algorithm. But the output is a vector.